src/app/filter-folder.pipe.ts

```diff
@@ -0,0 +1,22 @@
+import { ArrayType } from '@angular/compiler';
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'filterFolder'
+})
+export class FilterFolderPipe implements PipeTransform {
+
+  transform(data: any, search_name:string) {
```

Comment:
you need to add an additional check of the condition when the data will be empty.